velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -284,8 +286,20 @@ std::unique_ptr<ParquetTypeWithId> ReaderBase::getParquetColumnInfo(
       if (isFileColumnNamesReadAsLowerCase()) {
         folly::toLowerAscii(childName);
       }
-      auto childRequestedType =
-          requestedType ? requestedType->asRow().findChild(childName) : nullptr;
+
+      TypePtr childRequestedType = nullptr;
+      if (requestedType) {
+        if (requestedType->isRow()) {
+          auto fileTypeIdx =
+              requestedType->asRow().getChildIdxIfExists(childName);
+          if (fileTypeIdx.has_value()) {
+            childRequestedType = requestedType->asRow().childAt(*fileTypeIdx);
+          }
+        } else if (requestedType->isArray() || requestedType->isMap()) {
+          childRequestedType = requestedType;
```

Comment:
`childRequestedType` should be the element/key/value type

Reply:
This is indeed the case in normal situations, but when `requestedType` is `Array` or `Map`, when recursively processing child elements, I don't know whether it is a child element of `Array` or `Map`. So here I directly set `childRequestedType = requestedType`, and then later https://github.com/facebookincubator/velox/blob/15d5ae45129a61d75780f1fdc01cb9c8686a959e/velox/dwio/parquet/reader/ParquetReader.cpp#L400-L404 and https://github.com/facebookincubator/velox/blob/15d5ae45129a61d75780f1fdc01cb9c8686a959e/velox/dwio/parquet/reader/ParquetReader.cpp#L474-L481 process the child elements of Array or Map

Reply:
Sounds like we are not doing something correctly here.  How much change would it need to process array element/map key/value in its own recursive call, just like struct child?

Reply:
The `ReaderBase::getParquetColumnInfo` method needs to be modified to mark whether the parent type of the current processing type is array or map.

Reply:
Is adding a `const TypePtr& parentRequestedType` enough?

Reply:
@wypb can you address this review comment so that we can get this merged?

Reply:
@Yuhta @majetideepak I have introduced `parentRequestedType` according to the review comments. In local testing, you can get the child elements of Array/Map in this way. please help me review it, thank you.